examples/example.hpp

```diff
@@ -84,24 +88,117 @@ struct rect
+
+    void show()
+    {
+        ImGui::SetNextWindowSize(_size);
+        ImGui::SetNextWindowPos(_position);
+        //concate the name given with seq_id in order to make a unique name (needed for Begin())
+        std::string name_id = std::string(_name) + std::to_string(_seq_id);
+        ImGui::Begin(name_id.c_str(), nullptr, _sliders_flags);
+        std::string text_inside_slider = std::string(_name) + ": %.3f";
+        bool is_changed =
+            ImGui::SliderFloat("", &_value, _range.min, _range.max, text_inside_slider.c_str(), 5.0f, false); //logarithmic scale 
+        if (is_changed) {
+            _depth_sensor.set_option(RS2_OPTION_SEQUENCE_ID, _seq_id);
+            _depth_sensor.set_option(_option, _value);
+        }
+
+        ImGui::End();
+    }
+
+public:
+    const char* _name;
+    int _seq_id;
+    float _value;
+    rs2::option_range _range;
+    rs2::depth_sensor _depth_sensor;
+    rs2_option _option;
+    ImVec2 _position;
+    ImVec2 _size;
+    //flags for the sliders
+    int _sliders_flags = ImGuiWindowFlags_NoCollapse
+        | ImGuiWindowFlags_NoScrollbar
+        | ImGuiWindowFlags_NoSavedSettings
+        | ImGuiWindowFlags_NoResize
+        | ImGuiWindowFlags_NoMove
+        | ImGuiWindowFlags_NoTitleBar
+        | ImGuiWindowFlags_NoBringToFrontOnFocus;
+};
+
+//text box for ImGui
+class text_box {
+public:
+    text_box(const char* name, ImVec2 position, ImVec2 size) : _name(name), _position(position), _size(size) {}
+
+    void show(const char* text)
+    {
+        ImGui::SetNextWindowSize(_size);
+        ImGui::SetNextWindowPos(_position);
+        ImGui::Begin(_name, nullptr, _text_box_flags);
+        ImGui::Text(text);
+
+        ImGui::End();
+    }
+    void remove_title_bar() {
+        _text_box_flags |= ImGuiWindowFlags_NoTitleBar;
+    }
+public:
+    const char* _name;
+    ImVec2 _position;
+    ImVec2 _size;
+    // flags for displaying text box
+    int _text_box_flags = ImGuiWindowFlags_NoCollapse
+        | ImGuiWindowFlags_NoScrollbar
+        | ImGuiWindowFlags_NoSavedSettings
+        | ImGuiWindowFlags_NoResize
+        | ImGuiWindowFlags_NoMove
+        | ImGuiWindowFlags_NoFocusOnAppearing
+        | ImGuiWindowFlags_AlwaysUseWindowPadding
+        | ImGuiWindowFlags_NoBringToFrontOnFocus
+        | ImGuiWindowFlags_AlwaysAutoResize;
+};
+
+
 //////////////////////////////
 // Simple font loading code //
 //////////////////////////////
 
-#include "../third-party/stb_easy_font.h"
+//name aliasing the map of pairs<frame, tile_properties>
+using frame_and_tile_property = std::pair<rs2::frame, tile_properties>;
+using map_of_frames_and_tiles_properties = std::map<int, std::pair<rs2::frame, tile_properties>>;
 
-inline void draw_text(int x, int y, const char * text)
+
+inline void draw_text(int x, int y, const char* text)
 {
-    char buffer[60000]; // ~300 chars
+    std::vector<char>* buffer[60000]; 
```

Comment:
currently this vector allocates ptrs to chars - replace with vector of <char> without *.
Whenever this should be passed as parameter to function use buffer.data() or similar